contracts/Strategy.sol

```diff
@@ -617,12 +575,11 @@ contract Strategy is BaseStrategy {
 
             // request start of cooldown period
             uint256 cooldownStartTimestamp =
-            IStakedAave(stkAave).stakersCooldowns(address(this));
-            uint256 COOLDOWN_SECONDS = IStakedAave(stkAave).COOLDOWN_SECONDS();
-            uint256 UNSTAKE_WINDOW = IStakedAave(stkAave).UNSTAKE_WINDOW();
-            if (IERC20(address(stkAave)).balanceOf(address(this)) > 0 &&
-                (cooldownStartTimestamp == 0) ||
-                block.timestamp > cooldownStartTimestamp.add(COOLDOWN_SECONDS).add(UNSTAKE_WINDOW)) {
+                IStakedAave(stkAave).stakersCooldowns(address(this));
+            if (
+                IERC20(address(stkAave)).balanceOf(address(this)) > 0 &&
+                (((cooldownStartTimestamp == 0)) || _checkCooldown())
```

Comment:
my fault: 

I believe this is wrong. 

if the unstaking window passes, we get our rewards locked. 

we should revert to this:
```

if (IERC20(address(stkAave)).balanceOf(address(this)) > 0 &&
                ((cooldownStartTimestamp == 0) ||
                block.timestamp > cooldownStartTimestamp.add(COOLDOWN_SECONDS).add(UNSTAKE_WINDOW)))
```

Reply:
this was a try to get bytecode under limit. not needed anymore.

Reply:
I think I don't have access to the repo:

please replace: 
            ```
// request start of cooldown period
            uint256 cooldownStartTimestamp =
                IStakedAave(stkAave).stakersCooldowns(address(this));
            if (
                IERC20(address(stkAave)).balanceOf(address(this)) > 0 &&
                (((cooldownStartTimestamp == 0)) || _checkCooldown())
            ) {
                stkAave.cooldown();
            }
```

for 

```
// request start of cooldown period
            uint256 cooldownStartTimestamp =
            IStakedAave(stkAave).stakersCooldowns(address(this));
            uint256 COOLDOWN_SECONDS = IStakedAave(stkAave).COOLDOWN_SECONDS();
            uint256 UNSTAKE_WINDOW = IStakedAave(stkAave).UNSTAKE_WINDOW();
            if (IERC20(address(stkAave)).balanceOf(address(this)) > 0 &&
                (cooldownStartTimestamp == 0) ||
                block.timestamp > cooldownStartTimestamp.add(COOLDOWN_SECONDS).add(UNSTAKE_WINDOW)) {
                stkAave.cooldown();
            }
```

Reply:
@jmonteer I'm afraid this change breaks several tests. When harvesting the strategy it throws `INVALID_BALANCE_ON_COOLDOWN`. Tests in `test_debt_ratio.py` or `test_direct_transfer` are some examples to observe this behavior. I've invited you as a collaborator to my fork so you have commit access.

Reply:
@jmonteer this issue seems to be caused by a call to `cooldown()` with no balance staked (see https://github.com/aave/aave-stake-v2/blob/master/contracts/stake/StakedTokenV2.sol#L178). 

Because the `if` parenthesis is only grouping `cooldownStartTimestamp == 0`, the `||` condition is always evaluated to true and a call to `cooldown()` goes through even if the balance is zero.

I believe it should be changed from:

```
if (IERC20(address(stkAave)).balanceOf(address(this)) > 0 &&
                (cooldownStartTimestamp == 0) ||
                block.timestamp > cooldownStartTimestamp.add(COOLDOWN_SECONDS).add(UNSTAKE_WINDOW)
            )
```

to:

```
if (IERC20(address(stkAave)).balanceOf(address(this)) > 0 &&
                (cooldownStartTimestamp == 0 ||
                block.timestamp > cooldownStartTimestamp.add(COOLDOWN_SECONDS).add(UNSTAKE_WINDOW))
            )
```

This makes all tests pass. Let me know if this captures what you were trying to accomplish.

Reply:
that is exactly right! thanks

Reply:
in fact, that is what I put in the first comment, this morning writing it here I forgot about it but that's what tests are for... right? :D

thanks mono

Reply:
Awesome! Updated in 64997fbb2de4b29aef687220307f2b6fcbc00543.